### PR TITLE
[dashboard] Add alert when user deleting root level tab

### PR DESCRIPTION
**Pull request by graceguo-supercat**
one user reported that while she was editing dashboard, she accidentally deleted a root-level tab, but she didn't notice the action. She saved all the changes,  but later she found the loss is big.
So we want to add a alert when user deleting a root level tab. The tab will be removed after user confirm. The removed tab can still be recovered from undo.

**Step 1**
screen shot 2018-08-29 at 11 08 05 am

**Step 2**
before: tab is deleted.
after:
screen shot 2018-09-04 at 10 57 50 am

### CODE REVIEW

**Comment by williaster** [comment]
@graceguo-supercat great change!

A couple of aesthetic comments:
- there looks like there's double padding between the buttons, could we cut it in half?
- could we update the messaging to the following (minor tweaks, I don't think the warning/note keywords are needed since we're using a modal):

"""
**Delete dashboard tab?**

Deleting a tab will remove all content within it. You may still reverse this action with the **undo** button (...+ z) until you save your changes.
"""

File: superset/assets/src/dashboard/components/DeleteComponentModal.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+      <ModalTrigger
+        ref={this.setModalRef}
+        triggerNode={this.props.triggerNode}
+        modalTitle={t('Delete dashboard tab?')}
+        modalBody={
+          <div className="delete-component-modal">
+            <div>
+              <p />
+              <b>Warning:</b> If you delete this dashboard tab all content
```
Comment (williaster) [inline superset/assets/src/dashboard/components/DeleteComponentModal.jsx]: see my comments about updated messaging

File: superset/assets/src/dashboard/components/IconButton.jsx [hunk 2]
```diff
@@ -2,14 +2,15 @@ import React from 'react';
...
-  onClick: PropTypes.func.isRequired,
+  onClick: PropTypes.func,
   className: PropTypes.string,
   label: PropTypes.string,
 };
 
 const defaultProps = {
   className: null,
   label: null,
+  onClick: () => true,
```
Comment (williaster) [inline superset/assets/src/dashboard/components/IconButton.jsx]: hmm this is a little strange, but I guess is okay. Is this preferred so you don't have to set styles on a simple `<div className="fa fa-trash" />`?

File: superset/assets/src/dashboard/components/menu/WithPopoverMenu.jsx [hunk 3]
```diff
@@ -36,19 +37,19 @@ class WithPopoverMenu extends React.PureComponent {
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.editMode && nextProps.isFocused && !this.state.isFocused) {
-      document.addEventListener('click', this.handleClick, true);
```
Comment (williaster) [inline superset/assets/src/dashboard/components/menu/WithPopoverMenu.jsx]: yeah still can't remember why I added these 🤷‍♀️

File: superset/assets/src/dashboard/stylesheets/dashboard.less [hunk 4]
```diff
@@ -129,10 +129,25 @@ body {
...
+    width: 50px;
+    margin: 0;
+    position: relative;
+  }
+
+  .delete-modal-actions-container {
+    margin: 40px 0 12px 0;
+
+    .btn {
+      margin-right: 24px;
```
Comment (williaster) [inline superset/assets/src/dashboard/stylesheets/dashboard.less]: could we cut this in half?

**Review by williaster — APPROVED** [review]
This looks great to me with the exception of the messaging/aesthetic tweaks I suggested!

**Comment by elibrumbaugh** [comment]
Thanks for your aestheic suggestions @williaster! 

@graceguo-supercat could we remove the `command symbol` and instead write `cmd`? After thinking it over I'm worried that this will be confusing for Windows users. 

The command `cmd` key on Mac is widely understood to be `control` on Windows. However the symbol for command is not widely understood.

File: superset/assets/src/dashboard/components/IconButton.jsx (hunk 2 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/components/IconButton.jsx]: i want to re-use `IconButton` component, and it required a onClick callback. in my case, i want to use IconButton as trigger to show modal, and ModalTrigger will bind onClick itself. I can't pass in onClick callback for this instance.

so here i changed onClick prop not required, and add a default callback. Not sure this is right way.

File: superset/assets/src/dashboard/stylesheets/dashboard.less (hunk 4 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/stylesheets/dashboard.less]: i fixed padding, message text, and a few modal CSS. please see the updated attachment.

**Review by williaster — APPROVED** [review]
looks great, thanks @graceguo-supercat 🙌 🚢

**Pull request merged**